src/javascript/tests/lists.rs

```diff
@@ -0,0 +1,84 @@
+use super::*;
+use crate::assert_js;
+
+#[test]
+fn list_literals() {
+    assert_js!(
+        r#"
+fn go(x) {
+    []
+    [1]
+    [1, 2]
+    [1, 2, ..x]
+}
+"#,
+        r#""use strict";
+
+function go(x) {
+  [];
+  [1, []];
+  [1, [2, []]];
+  return [1, [2, x]];
+}
+"#
+    );
+}
+
+#[test]
+fn long_list_literals() {
+    assert_js!(
+        r#"
+fn go() {
+    [111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111]
+    [11111111111111111111111111111111111111111111, 11111111111111111111111111111111111111111111111111111111111111111111111111111111]
+}
+"#,
+        r#""use strict";
+
+function go() {
+  [111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111,
+    []];
+  return [11111111111111111111111111111111111111111111,
+    [11111111111111111111111111111111111111111111111111111111111111111111111111111111,
```

Comment:
I'm reasonably happy with this formatting. gets a bit messy with multiline elements but I think it was always going too

Reply:
Can we go for this kind of formatting?

```js
return [
  11111111111111111111111111111111111111111111,
  [
    11111111111111111111111111111111111111111111111111111111111111111111111111111111
    []
  ],
]
```

I think that's a bit more "javascripty". We use this formatting for calls so can copy the structure of the document from there.

Reply:
I can try, I did go for something like this but each new element had another indent call so my first version wasn't working.

At this point I'm not sure any approach is very javascripty, and I liked the compactness of this version but I can give a crack at it.

Reply:
Actually yeah that's a good point. Let's use this, at least for now